src/commands/generateApiFromCode.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { IActionContext } from "@microsoft/vscode-azext-utils";
+import * as fs from "fs";
+import * as vscode from 'vscode';
+import { ExceedTokenLimit, MODEL_SELECTOR } from "../constants";
+import { UiStrings } from "../uiStrings";
+import { GeneralUtils } from "../utils/generalUtils";
+
+export namespace GenerateApiFromCode {
+    export async function generateApiFromCode(context: IActionContext) {
+        const activeEditor = vscode.window.activeTextEditor;
+
+        if (!activeEditor) {
+            throw new Error(UiStrings.NoActiveFileOpen);
+        }
+
+        const languageId = activeEditor.document.languageId;
+        const fileUri = activeEditor.document.uri;
+
+        await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: UiStrings.GeneratingOpenAPI
+        }, async (progress, token) => {
+            try {
+                const codeContent = await fs.promises.readFile(fileUri.fsPath, { encoding: 'utf-8' });
+
+                const messages = [
+                    vscode.LanguageModelChatMessage.User(`You are an expert in ${languageId} programming language and OpenAPI.

```

Comment:
move to `copilot-chat/constants.ts` file ?

Reply:
This is not a simple constant, but prompt with parameters.  In furure, I will consider adpoting https://www.npmjs.com/package/@vscode/prompt-tsx. So no change for now.